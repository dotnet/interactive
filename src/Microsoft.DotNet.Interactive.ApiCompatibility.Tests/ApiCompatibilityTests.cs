// Copyright (c) .NET Foundation and contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System.Diagnostics;
using Assent;
using Microsoft.DotNet.Interactive.CSharp;
using Microsoft.DotNet.Interactive.Documents;
using Microsoft.DotNet.Interactive.Formatting;
using Microsoft.DotNet.Interactive.FSharp;
using Microsoft.DotNet.Interactive.Journey;
using Microsoft.DotNet.Interactive.Kql;
using Microsoft.DotNet.Interactive.PowerShell;
using Microsoft.DotNet.Interactive.SqlServer;
using Microsoft.DotNet.Interactive.Tests.Utility;
using Xunit;

namespace Microsoft.DotNet.Interactive.ApiCompatibility.Tests;

public class ApiCompatibilityTests
{
    private readonly Configuration _configuration;

    public ApiCompatibilityTests()
    {
        _configuration = new Configuration()
                         .SetInteractive(Debugger.IsAttached)
                         .UsingExtension("txt");
    }

    [FactSkipLinux]
    public void Interactive_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<Kernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void Formatting_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<FormatContext>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void Document_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<InteractiveDocument>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void PackageManagement_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<PackageRestoreContext>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void Journey_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<Lesson>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void csharp_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<CSharpKernel>();
        this.Assent(contract, _configuration);
    }

    [Fact(Skip = "need to use signature files")]
    public void fsharp_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<FSharpKernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void powershell_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<PowerShellKernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void mssql_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<MsSqlKernelConnector>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void kql_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<KqlKernelConnector>();
        this.Assent(contract, _configuration);
    }
}