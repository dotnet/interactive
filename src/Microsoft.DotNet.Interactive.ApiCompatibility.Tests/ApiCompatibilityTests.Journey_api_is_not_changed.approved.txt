Microsoft.DotNet.Interactive.Journey
  public class Challenge
    .ctor(System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SubmitCode> setup = null, System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SendEditableCode> contents = null, System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SubmitCode> environmentSetup = null, System.String name = null)
    public System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SendEditableCode> Contents { get;}
    public ChallengeEvaluation CurrentEvaluation { get;}
    public ChallengeSubmission CurrentSubmission { get;}
    public System.Func<ChallengeContext,System.Threading.Tasks.Task> DefaultProgressionHandler { get; set;}
    public System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SubmitCode> EnvironmentSetup { get;}
    public System.Boolean IsSetup { get;}
    public Lesson Lesson { get;}
    public System.String Name { get;}
    public System.Func<ChallengeContext,System.Threading.Tasks.Task> OnCodeSubmittedHandler { get;}
    public System.Boolean Revealed { get;}
    public System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SubmitCode> Setup { get;}
    public System.Collections.Generic.IEnumerable<ChallengeSubmission> SubmissionHistory { get;}
    public System.Void AddRule(System.String name, System.Action<RuleContext> action)
    public System.Void AddRule(System.Action<RuleContext> action)
    public System.Void AddRuleAsync(System.String name, System.Func<RuleContext,System.Threading.Tasks.Task> action)
    public System.Void AddRuleAsync(System.Func<RuleContext,System.Threading.Tasks.Task> action)
    public System.Threading.Tasks.Task Evaluate(System.String submittedCode = null, System.Collections.Generic.IEnumerable<Microsoft.DotNet.Interactive.Events.KernelEvent> events = null)
    public System.Threading.Tasks.Task InvokeOnCodeSubmittedHandler()
    public System.Void OnCodeSubmitted(System.Action<ChallengeContext> action)
    public System.Void OnCodeSubmittedAsync(System.Func<ChallengeContext,System.Threading.Tasks.Task> action)
  public class ChallengeContext
    .ctor(Challenge challenge)
    public Challenge Challenge { get;}
    public ChallengeEvaluation Evaluation { get;}
    public Lesson Lesson { get;}
    public System.Collections.Generic.IEnumerable<RuleEvaluation> RuleEvaluations { get;}
    public System.Collections.Generic.IEnumerable<ChallengeSubmission> SubmissionHistory { get;}
    public System.Void SetMessage(System.String message, System.Object hint = null)
    public System.Threading.Tasks.Task StartChallengeAsync(Challenge challenge)
    public System.Threading.Tasks.Task StartChallengeAsync(System.String name)
    public System.Threading.Tasks.Task StartNextChallengeAsync()
  public class ChallengeDefinition
    .ctor(System.String name, System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SubmitCode> setup, System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SendEditableCode> contents, System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SubmitCode> environmentSetup)
    public System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SendEditableCode> Contents { get;}
    public System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SubmitCode> EnvironmentSetup { get;}
    public System.String Name { get;}
    public System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SubmitCode> Setup { get;}
    public Challenge ToChallenge()
  public enum ChallengeDirective : System.Enum, System.IComparable, System.IConvertible, System.IFormattable
    ChallengeSetup=0
    Question=1
    Scratchpad=2
  public class ChallengeEvaluation
    .ctor(System.String label = null)
    public System.Object Hint { get;}
    public System.String Message { get;}
    public System.Collections.Generic.IEnumerable<RuleEvaluation> RuleEvaluations { get;}
    public System.Void SetMessage(System.String message, System.Object hint = null)
    public System.Void SetRuleOutcome(System.String name, Outcome outcome, System.String reason = null, System.Object hint = null)
  public static class ChallengeExtensions
    public static Challenge SetDefaultProgressionHandler(Challenge nextChallenge)
    public static System.Void SetDefaultProgressionHandlers()
  public class ChallengeSubmission
    .ctor(System.String submittedCode, ChallengeEvaluation evaluation, System.Collections.Generic.IEnumerable<Microsoft.DotNet.Interactive.Events.KernelEvent> events)
    public ChallengeEvaluation Evaluation { get;}
    public System.Collections.Generic.IEnumerable<Microsoft.DotNet.Interactive.Events.KernelEvent> EventsProduced { get;}
    public System.Collections.Generic.IEnumerable<RuleEvaluation> RuleEvaluations { get;}
    public System.String SubmittedCode { get;}
  public static class KernelExtensions
    public static System.Threading.Tasks.Task InitializeChallenge(Challenge challengeToInit)
    public static Microsoft.DotNet.Interactive.CompositeKernel UseModelAnswerValidation()
    public static Microsoft.DotNet.Interactive.CompositeKernel UseProgressiveLearning(System.Net.Http.HttpClient httpClient = null)
    public static Microsoft.DotNet.Interactive.CompositeKernel UseProgressiveLearningMiddleware()
  public class Lesson
    public static Challenge CurrentChallenge { get;}
    public static LessonMode Mode { get; set;}
    public static System.String Name { get;}
    public static System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SubmitCode> Setup { get;}
    public static System.Void Clear()
    public static System.Void From(LessonDefinition definition)
    public static System.Boolean IsSetupCommand(Microsoft.DotNet.Interactive.Commands.KernelCommand command)
    public static System.Void ResetChallenge()
    public static System.Void SetChallengeLookup(System.Func<System.String,Challenge> handler)
    public static System.Void SetChallengeLookup(System.Func<System.String,System.Threading.Tasks.Task<Challenge>> handler)
    public static System.Threading.Tasks.Task StartChallengeAsync(Challenge challenge)
    public static System.Threading.Tasks.Task StartChallengeAsync(System.String name)
    .ctor()
  public class LessonDefinition
    .ctor(System.String name, System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SubmitCode> setup)
    public System.String Name { get;}
    public System.Collections.Generic.IReadOnlyList<Microsoft.DotNet.Interactive.Commands.SubmitCode> Setup { get;}
  public enum LessonDirective : System.Enum, System.IComparable, System.IConvertible, System.IFormattable
    Challenge=0
  public enum LessonMode : System.Enum, System.IComparable, System.IConvertible, System.IFormattable
    TeacherMode=0
    StudentMode=1
  public static class Main
    public static System.Threading.Tasks.Task OnLoadAsync(Microsoft.DotNet.Interactive.Kernel kernel, System.Net.Http.HttpClient httpClient = null)
  public class NotebookLessonParser
    public static System.Collections.Generic.List<System.String> AllDirectiveNames { get;}
    public static System.Threading.Tasks.Task<Microsoft.DotNet.Interactive.Documents.InteractiveDocument> LoadNotebookFromUrl(System.Uri uri, System.Net.Http.HttpClient httpClient = null, Microsoft.DotNet.Interactive.CompositeKernel kernel = null)
    public static System.Void Parse(Microsoft.DotNet.Interactive.Documents.InteractiveDocument document, ref LessonDefinition& lesson, ref System.Collections.Generic.List<ChallengeDefinition> challenges)
    public static System.Threading.Tasks.Task<Microsoft.DotNet.Interactive.Documents.InteractiveDocument> ReadFileAsInteractiveDocument(System.IO.FileInfo file, Microsoft.DotNet.Interactive.CompositeKernel kernel = null)
    .ctor()
  public enum Outcome : System.Enum, System.IComparable, System.IConvertible, System.IFormattable
    Failure=0
    PartialSuccess=1
    Success=2
  public class Rule
    .ctor(System.Func<RuleContext,System.Threading.Tasks.Task> ruleConstraints, System.String name = null)
    public System.String Name { get;}
  public class RuleContext
    .ctor(ChallengeContext challengeContext, System.String submittedCode = null, System.Collections.Generic.IEnumerable<Microsoft.DotNet.Interactive.Events.KernelEvent> events = null, System.String name = )
    public Challenge Challenge { get;}
    public System.Collections.Generic.IEnumerable<Microsoft.DotNet.Interactive.Events.KernelEvent> EventsProduced { get;}
    public System.String Name { get;}
    public System.Boolean Passed { get;}
    public System.String SubmittedCode { get;}
    public System.Void Fail(System.String reason = null, System.Object hint = null)
    public System.Void PartialPass(System.String reason = null, System.Object hint = null)
    public System.Void Pass(System.String reason = null, System.Object hint = null)
  public class RuleEvaluation
    .ctor(Outcome outcome, System.String name = null, System.String reason = null, System.Object hint = null)
    public System.Object Hint { get;}
    public System.String Name { get;}
    public Outcome Outcome { get;}
    public System.Boolean Passed { get;}
    public System.String Reason { get;}
Microsoft.DotNet.Interactive.Journey.Utilities
  public class OrderedTable<T>, ICollection<T>, IEnumerable<T>, System.Collections.IEnumerable
    .ctor()
    .ctor(IEqualityComparer<T> comparer)
    public System.Int32 Count { get;}
    public System.Boolean IsReadOnly { get;}
    public System.Boolean Add(T item)
    public System.Void Clear()
    public System.Boolean Contains(T item)
    public System.Void CopyTo(T[] array, System.Int32 arrayIndex)
    public IEnumerator<T> GetEnumerator()
    public System.Boolean Remove(T item)
